src/transformers/models/vitpose/modeling_vitpose.py

```diff
@@ -0,0 +1,294 @@
+    as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage and
+    behavior.
+
+    Parameters:
+        config ([`ViTPoseConfig`]): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+VITPOSE_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Pixel values. Pixel values can be obtained using [`ViTPoseImageProcessor`]. See
+            [`ViTPoseImageProcessor.__call__`] for details.
+
+        head_mask (`torch.FloatTensor` of shape `(num_heads,)` or `(num_layers, num_heads)`, *optional*):
+            Mask to nullify selected heads of the self-attention modules. Mask values selected in `[0, 1]`:
+
+            - 1 indicates the head is **not masked**,
+            - 0 indicates the head is **masked**.
+
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+def flip_back(output_flipped, flip_pairs, target_type="GaussianHeatmap"):
+    """Flip the flipped heatmaps back to the original form.
+
+    Args:
+        output_flipped (`np.ndarray` of shape `(batch_size, num_keypoints, height, width)`):
+            The output heatmaps obtained from the flipped images.
+        flip_pairs (list[tuple()):
+            Pairs of keypoints which are mirrored (for example, left ear -- right ear).
+        target_type (str):
+            GaussianHeatmap or CombinedTarget
+
+    Returns:
+        np.ndarray: heatmaps that flipped back to the original image
+    """
+    if output_flipped.ndim != 4:
+        raise ValueError("output_flipped should be [batch_size, num_keypoints, height, width]")
+    shape_ori = output_flipped.shape
+    channels = 1
+    if target_type.lower() == "CombinedTarget".lower():
+        channels = 3
+        output_flipped[:, 1::3, ...] = -output_flipped[:, 1::3, ...]
+    output_flipped = output_flipped.reshape(shape_ori[0], -1, channels, shape_ori[2], shape_ori[3])
+    output_flipped_back = output_flipped.copy()
+
+    # Swap left-right parts
+    for left, right in flip_pairs.tolist():
+        output_flipped_back[:, left, ...] = output_flipped[:, right, ...]
+        output_flipped_back[:, right, ...] = output_flipped[:, left, ...]
```

Comment:
Is this correct? This doesn't flip, it makes everything `output_flipped[:, right, ...]`

Reply:
I took this from here: https://github.com/ViTAE-Transformer/ViTPose/blob/d5216452796c90c6bc29f5c5ec0bdba94366768a/mmpose/core/post_processing/post_transforms.py#L140-L144. Could you clarify why this doesn't flip?

Reply:
I think it is flip sine it is `output_flipped_back` not `output_flipped`. @amyeroberts Can you explain more if it not simple confuse?

```
import torch
a = [[[0],[1]]]
a = torch.tensor(a)
print(a)
b = a.clone()
b[:, 1, ...] = a[:, 0, ...]
b[:, 0, ...] = a[:, 1, ...]
print(b)

tensor([[[0],
         [1]]])
tensor([[[1],
         [0]]])
```

Reply:
You're right, I was misreading it as `output_flipped[:, left, ...] = output_flipped[:, right, ...]`